File: api/src/endpoint/repository/get_repository_commit.rs
use api_derive::ApiRequest;
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::repository::RepositoryCommitResource};

pub struct GetRepositoryCommit;

impl Endpoint for GetRepositoryCommit {
    const PATH: &'static str = "/repository/{owner}/{repo}/commits/{sha}";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetRepositoryCommitRequest;
    type Response = GetRepositoryCommitResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetRepositoryCommitRequest {}

pub type GetRepositoryCommitResponse = RepositoryCommitResource;

File: gitdot-api-ts/src/endpoint/repository/get-repository-commit.ts
import { z } from "zod";

import { RepositoryCommitResource } from "../../resource";

export const GetRepositoryCommitRequest = z.object({});
export type GetRepositoryCommitRequest = z.infer<
  typeof GetRepositoryCommitRequest
>;

export const GetRepositoryCommit = {
  path: "/repository/{owner}/{repo}/commits/{sha}",
  method: "GET",
  request: GetRepositoryCommitRequest,
  response: RepositoryCommitResource,
} as const;
export type GetRepositoryCommit = typeof GetRepositoryCommit;
